pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1203,6 +1234,44 @@ func (m *kubeGenericRuntimeManager) computePodActions(ctx context.Context, pod *
 	return changes
 }
 
+// getContainersToRemove returns container info about the containers to remove from the runtime.
+// The first list are the containers that triggered the RestartAllContainers; the second list
+// are the containers that are victim of the RestartAllContainers.
+func (m *kubeGenericRuntimeManager) getContainersToRemove(containers []v1.Container, podStatus *kubecontainer.PodStatus) (sources []containerToRemoveInfo, targets []containerToRemoveInfo) {
+	// Copy the slice before sorting it
+	containerStatusesCopy := make([]*kubecontainer.Status, len(podStatus.ContainerStatuses))
+	copy(containerStatusesCopy, podStatus.ContainerStatuses)
+	// Make the latest container status comes first.
+	sort.Sort(sort.Reverse(kubecontainer.SortContainerStatusesByCreationTime(containerStatusesCopy)))
+	for idx, c := range containers {
+		isLatest := true
+		for _, containerStatus := range podStatus.ContainerStatuses {
+			if containerStatus.Name != c.Name {
```

Comment:
nit: add a comment here that there can be multiple container statuses per container, which is why you can't use FindContainerStatusByName